File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>

int CountLetters(string text);
int CountWords(string text);
int CountSentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    int nLetters = 0, nWords = 0, nSentences = 0;

    nLetters = CountLetters(text);
    nWords = CountWords(text);
    nSentences = CountSentences(text);

    /*
    printf("%d\n", nLetters);
    printf("%d\n", nWords);
    printf("%d\n", nSentences);
    */

    float L = (nLetters / (float) nWords) * 100;
    float S = (nSentences / (float) nWords) * 100;

    /*
    printf("L: %f\n", L);
    printf("S: %f\n", S);
    */

    float grade = round(0.0588 * L - 0.296 * S - 15.8);

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", grade);
    }

    return 0;
}

int CountLetters(string s)
{
    int count = 0;
    for (int i = 0; s[i] != '\0'; i++)
    {
        if (isalpha(s[i]))
        {
            count++;
        }
    }

    return count;
}

int CountWords(string s)
{
    int count = 0;
    for (int i = 0; s[i] != '\0'; i++)
    {
        if (isspace(s[i]))
        {
            count++;
        }
    }
    count++; // Because every word has a space before it, except the first one

    return count;
}

int CountSentences(string s)
{
    int count = 0;
    for (int i = 0; s[i] != '\0'; i++)
    {
        if (s[i] == '.' || s[i] == '!' || s[i] == '?')
        {
            count++;
        }
    }

    return count;
}

File: sentimental-readability/readability.py
from cs50 import get_string

def main():
    text = get_string()
    nLetters = nWords = nSentences = 0

    nLetters = CountLetters(text)
    nWords = CountWords(text)
    nSentences = CountSentences(text)

    L = (nLetters / nWords) * 100
    S = (nSentences / nWords) * 100
    grade = round(0.0588 * L - 0.296 * S - 15.8)

    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")

    exit(0)

def CountLetters(text: str) -> int:
    count = 0
    for letter in text:
        if letter.isalpha():
            count += 1
    return count

def CountWords(text: str) -> int:
    count = 0
    for letter in text:
        if letter.isspace():
            count += 1
    count += 1 # Because every word has a space before it except the first one
    return count

def CountSentences(text: str) -> int:
    count = 0
    for letter in text:
        if letter in [".", "?", "!"]:
            count += 1
    return count

main()